src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -110,6 +110,13 @@ public class MergeBlockDownloader : BlockDownloader
 
             bool HasMoreToSync(out BlockHeader[]? headers, out int headersToRequest)
             {
+                if (!_beaconPivot.BeaconPivotExists())
```

Comment:
This condition is wrong. We can have situation when we don't have beacon pivot and we want to sync.

Reply:
what do you think about Math.Min(ProcessDestination, BestKnownNumber) >= BestKnownBeaconNumber?

Reply:
Imagine that you have timeout in NewPayloadHandler. For the next payload we insert it without cache and return SYNCING. We don't have beaconPivot in this situation.